spec.md

```diff
@@ -2070,103 +2098,20 @@ module Foo:
   define z[1] = probe(w)
 ```
 
-`RWProbe`{.firrtl} references to ports are not allowed on public-facing modules.
-
-Define statements can set a `Probe`{.firrtl} to either a `Probe`{.firrtl} or `RWProbe`{.firrtl}, but a `RWProbe`{.firrtl} cannot be set to a `Probe`{.firrtl}.
-The inner types of the two references must (recursively) be identical or identical with the destination containing uninferred versions of the corresponding element in the source type.
-See [@sec:type-inference] for details.
-
-### Probes and Passive Types
-
-While `Probe`{.firrtl} inner types are passive, the type of the probed static reference is not required to be:
-
-``` firrtl
-module Foo :
-  input x : {a: UInt, flip b: UInt}
-  output y : {a: UInt, flip b: UInt}
-  output xp : Probe<{a: UInt, b: UInt}> ; passive
-
-  wire p : {a: UInt, flip b: UInt} ; p is not passive
-  define xp = probe(p)
-  connect p, x
-  connect y, p
-```
-
-### Probes and Const Types
-
-Probe types may target `const`{.firrtl} signals, but cannot use `rwprobe`{.firrtl} with a constant signal to produce a `RWProbe<const T>`{.firrtl}.
-
-### Exporting References to Nested Declarations
-
-Nested declarations (see [@sec:nested-declarations]) may be exported:
-
-``` firrtl
-module RefProducer :
-  input a : UInt<4>
-  input en : UInt<1>
-  input clk : Clock
-  output thereg : Probe<UInt>
-
-  when en :
-    reg myreg : UInt, clk
-    connect myreg, a
-    define thereg = probe(myreg)
-```
-
-### Forwarding References Upwards
-
-Define statements can be used to forward a child module's reference further up the hierarchy:
-
-``` firrtl
-module Foo :
-  output p : Probe<UInt>
-  ; ...
-
-module Forward :
-  output p : Probe<UInt>
-
-  inst f of Foo
-  define p = f.p
-```
-
-Define statements may narrow a probe of an aggregate to a sub-element using static expression:
+## The `read`{.firrtl} Expressions
 
-``` firrtl
-module Foo :
-  output p : Probe<UInt[2]>[2]
-  ; ...
+Both `Probe`{.firrtl} and `RWProbe`{.firrtl} may be read from using the `read`{.firrtl} expression.
 
-module Forward :
-  output p : Probe<UInt>
-
-  inst f of Foo
-  define p = f.p[0][1]
-```
+Keep in mind that probes are intended for verification and isn't intended to be synthesizable.
+A `read`{.firrtl} expression can reach arbitrarily far into the module hierarchy to read from a distant circuit component, which is arguably unphysical at worst and potentially expensive at best.
```

Comment:
all of this feels very... commentary-like. I'm not sure what it really has to do with the firrtl spec , especially without considering the one ABI.

Reply:
Agreed. However, I think it's good to have this as a placeholder until we can formalize the distinction between "synthesizable"/"hardware" and "verification-only"/"simulation-only".